controller/api/models.py

```diff
@@ -287,8 +288,9 @@ def _start_containers(self, to_add):
         [t.start() for t in create_threads]
         [t.join() for t in create_threads]
         if any(c.state != 'created' for c in to_add):
-            err = 'aborting, failed to create some containers'
+            err = 'aborting, failed to create some containers rolling back'
             log_event(self, err, logging.ERROR)
+            self._destroy_containers(to_add)
```

Comment:
@bacongobbler  release.destroy() has no link t0 destroy containers it might be doing some  cleaning or deleting the database records ?? I haven't looked much. But interestingly I have no idea why there are three different threads trying to scale when something fails  . Do we have any retry settings if scale fails ??
https://gist.github.com/smothiki/cd30977234cf7d1d3c7d